subsys/bluetooth/audio/has.c

```diff
@@ -508,26 +507,15 @@ static void bond_deleted_cb(uint8_t id, const bt_addr_le_t *addr)
 	if (context != NULL) {
 		context_free(context);
 	}
+
+	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		bt_settings_delete("has", 0, addr);
+	}
```

Comment:
We don't really have documentation for the internal `bt_settings` APIs; when setting `id=0` does that mean delete for any IDs, or specifically for `BT_ID_DEFAULT` (0)?

Reply:
I believe it deletes the one with the ID that was provided when the entry was added (0 in this case).

Reply:
Not necessarily something that needs to be fixed in this PR, but I think we should figure out and understand how the `0` ID value here is used in the `bt_settings`, and preferably also replace all the literal constant values used by the stack with a `#define` if it has special meaning.

Reply:
@jhedberg Looks like you and @Vudentz have been involved with the `bt_settings`, right?

Reply:
@jhedberg Do you know the intention of the `id` parameter in `bt_settings`?